test/ClientFastTests/Subscriptions/SubscriptionOperationsSignaling.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FastTests;
+using FastTests.Subscriptions;
+using Raven.NewClient.Abstractions.Data;
+using Raven.NewClient.Client.Exceptions.Subscriptions;
+using Raven.Tests.Core.Utils.Entities;
+using Xunit;
+
+namespace NewClientTests.NewClient.Subscriptions
+{
+    public class SubscriptionOperationsSignaling : RavenNewTestBase
+    {
+        private readonly TimeSpan _reasonableWaitTime = TimeSpan.FromSeconds(20);// todo: reduce to 20
+
+        [Fact]
+        public void WaitOnSubscriptionTaskWhenSubscriptionIsOvertaken()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var subscriptionId = store.Subscriptions.Create(new SubscriptionCriteria<User>());
+
+                var subscription = store.Subscriptions.Open<User>(new SubscriptionConnectionOptions(subscriptionId));
+                
+                var users = new BlockingCollection<User>();
+                
+                subscription.Subscribe(users.Add);
+                subscription.Start();
+
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new User());
+                    session.SaveChanges();
+                }
+
+                User user;
+                Assert.True(users.TryTake(out user, _reasonableWaitTime));
+
+                var concurrentSubscription = store.Subscriptions.Open<User>(
+                    new SubscriptionConnectionOptions(subscriptionId) { 
+                    Strategy = SubscriptionOpeningStrategy.TakeOver
+                });
+
+                var thread = new Thread(() => {
+                    Thread.Sleep(300);
```

Comment:
Can we do it without a dedicated thread and a sleep? A task with an event should also work here, no?